commands/healthCheck/health-check.js

```diff
@@ -0,0 +1,110 @@
+import { SlashCommandBuilder } from "discord.js";
+
+// ---- In-memory state for DB failures (resets when bot restarts) ----
+let consecutiveDbFailures = 0;
+
+// ---- Helper: build admin mentions from env ----
+function getAdminMentions() {
+  const raw = process.env.adminUserIds || "";
+  if (!raw) return "";
+  const ids = raw
+    .split(",")
+    .map((id) => id.trim())
+    .filter(Boolean);
+
+  if (ids.length === 0) return "";
+  return ids.map((id) => `<@${id}>`).join(" ");
+}
+
+// ---- Helper: safely log DB errors to the issue channel ----
+async function postDbErrorToIssuesChannel(interaction, error, isRepeated) {
+  const channelId = process.env.botIssuesChannelId;
+  if (!channelId) {
+    console.warn("[BOT-ISSUES] botIssuesChannelId is not set in .env");
+    return;
+  }
+
+  try {
+    const channel = await interaction.client.channels.fetch(channelId);
+    if (!channel || typeof channel.send !== "function") {
+      console.error("[BOT-ISSUES] Channel is not text-based or not sendable:", channelId);
+
+      return;
+    }
+
+    let content =
+      `⚠️ **DB health check failure**\n` +
+      `Guild: \`${interaction.guild?.id ?? "DM"}\`\n` +
+      `User: <@${interaction.user.id}> (${interaction.user.id})\n` +
+      `Error: \`${error.message}\``;
+
+    if (isRepeated) {
+      const adminMentions = getAdminMentions();
+      if (adminMentions) {
+        content += `\n\n🚨 **Repeated failures detected.** ${adminMentions}`;
+      }
+    }
+
+    await channel.send({ content });
+  } catch (err) {
+    // Never crash the command because of logging failures
+    console.error("[BOT-ISSUES ERROR] Failed to send DB error message:", err);
+  }
+}
```

Comment:
The helper functions (`getAdminMentions`, `postDbErrorToIssuesChannel`) don’t really belong inside the command. They’re utilities, so they could be moved into a small helper module and imported here. Keeps the command focused and avoids this file ballooning as more edge cases get bolted on in the future.